cscs-checks/system/io/ior_check.py

```diff
@@ -76,86 +87,88 @@ def __init__(self, base_dir):
 
-        self.ior_block_size = self.fs[base_dir]['ior_block_size']
-        self.ior_access_type = self.fs[base_dir]['ior_access_type']
-        self.executable_opts = ['-B', '-F', '-C ', '-Q 1', '-t 4m', '-D 30',
-                                '-b', self.ior_block_size,
-                                '-a', self.ior_access_type]
         self.sourcesdir = os.path.join(self.current_system.resourcesdir, 'IOR')
-        self.executable = os.path.join('src', 'C', 'IOR')
-        self.build_system = 'Make'
-
-        vpe = 'valid_prog_environs'
-        penv = self.fs[base_dir][cur_sys].get(vpe, ['builtin'])
-        self.valid_prog_environs = penv
 
+    @run_after('init')
+    def set_build_systems(self):
+        self.build_system = 'Make'
         self.build_system.options = ['posix', 'mpiio']
         self.build_system.max_concurrency = 1
         self.num_gpus_per_node = 0
 
+    @run_before('run')
+    def set_prerun_cmds(self):
         # Default umask is 0022, which generates file permissions -rw-r--r--
         # we want -rw-rw-r-- so we set umask to 0002
         os.umask(2)
-        self.time_limit = '5m'
-        # Our references are based on fs types but regression needs reference
-        # per system.
-        self.reference = {
-            '*': self.fs[base_dir]['reference']
-        }
-
-        self.maintainers = ['SO', 'GLR']
-
-        systems_to_test = ['dom', 'daint', 'eiger', 'pilatus']
-        if self.current_system.name in systems_to_test:
-            self.tags |= {'production', 'external-resources'}
+        self.test_dir = os.path.join(self.base_dir, self.username, '.ior')
+        self.prerun_cmds = ['mkdir -p ' + self.test_dir]
 
     @run_before('run')
     def set_exec_opts(self):
+        self.executable = os.path.join('src', 'C', 'IOR')
+        self.test_file = os.path.join(self.test_dir, 'ior')
         self.test_file += '.' + self.current_partition.name
-        self.executable_opts += ['-o', self.test_file]
+        # executable options depends on the file system
+        self.ior_block_size = self.fs[self.base_dir]['ior_block_size']
+        self.ior_access_type = self.fs[self.base_dir]['ior_access_type']
+        self.executable_opts = ['-B', '-F', '-C ', '-Q 1', '-t 4m', '-D 30',
+                                '-b', self.ior_block_size,
+                                '-a', self.ior_access_type,
+                                '-o', self.test_file]
 
 
-@rfm.parameterized_test(['/scratch/e1000'],
-                        ['/scratch/snx3000tds'],
-                        ['/scratch/snx3000'],
-                        ['/users'],
-                        ['/scratch/shared/fulen'])
+@rfm.simple_test
 class IorWriteCheck(IorCheck):
-    def __init__(self, base_dir):
-        super().__init__(base_dir)
-        self.executable_opts += ['-w', '-k']
+    executable_opts += ['-w', '-k']
+    tags |= {'write'}
+
+    @run_after('init')
+    def set_sanity_perf_patterns(self):
```

Comment:
I would split that and use the `@sanity_function` syntax for the sanity patterns.